openupgrade_scripts/scripts/loyalty/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,806 @@
+                WHEN applies_on = 'on_next_order' THEN 'future'
+            END
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET trigger =
+            CASE
+                WHEN trigger = 'no_code_needed' THEN 'auto'
+                WHEN trigger = 'code_needed' THEN 'with_code'
+            END
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET program_type =
+            CASE
+                WHEN program_type = 'coupon_program' THEN 'coupons'
+                WHEN program_type = 'promotion_program' THEN 'promotion'
+            END
+        """,
+    )
+
+
+def update_loyalty_reward_data(env):
+    # Determine which program_id and company_id each reward belongs to
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_reward
+        ADD COLUMN IF NOT EXISTS program_id INT
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_reward AS lr
+        SET program_id = lp.id
+        FROM loyalty_program AS lp
+        WHERE lr.id = lp.reward_id
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_reward
+        ADD COLUMN IF NOT EXISTS company_id INT
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_reward AS lr
+        SET company_id = lp.company_id
+        FROM loyalty_program AS lp
+        WHERE lr.id = lp.reward_id
+        """,
+    )
+    # Update discount values according to discount mode
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_reward
+        ADD COLUMN IF NOT EXISTS discount FLOAT
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_reward
+        SET discount = CASE
+            WHEN discount_mode = 'percentage' THEN discount_percentage
+            WHEN discount_mode = 'fixed_amount' THEN discount_fixed_amount
+        END
+        """,
+    )
+    # Establish whether a reward is active based on the status of the program to which the
+    # reward belongs
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_reward
+        ADD COLUMN IF NOT EXISTS active BOOLEAN
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_reward AS lr
+        SET active = lp.active
+        FROM loyalty_program AS lp
+        WHERE lr.id = lp.reward_id
+        """,
+    )
+    # Set default values
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_reward
+        ADD COLUMN IF NOT EXISTS clear_wallet BOOLEAN
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_reward
+        SET clear_wallet = false
```

Comment:
Instead of this, user `add_fields` from openupgradelib with a default value. (The same for others)